src/coinbase/user_operation.ts

```diff
@@ -0,0 +1,124 @@
+import { Coinbase } from "./coinbase";
+import { UserOperation as UserOperationModel, UserOperationStatusEnum } from "../client/api";
+import { delay } from "./utils";
+import { TimeoutError } from "./errors";
+
+/**
+ * A representation of a UserOperation, which calls a smart contract method
+ * onchain.
+ */
+export class UserOperation {
```

Comment:
this feels very ruby to me, all functions other than wait and reload are just a property on model or address. Might be worth just using functions.